ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/SqlLedger.scala

```diff
@@ -399,7 +409,7 @@ private class SqlLedgerFactory(ledgerDao: LedgerDao) {
                 logger.warn(
                   s"Initial packages provided, presumably as command line arguments, but I'm picking up from an existing database, and thus they will not be used")
               }
-              ledgerFound(foundLedgerId)
+              ledgerFound(foundLedgerId, participantId)
```

Comment:
The participant ID is checked in `ledgerFound`, the ledger ID is checked in directly in `initialize`. Can you write the `initialize` method as something like this?
```scala
Future.sequence((ledgerDao.lookupLedgerId(), ledgerDao.lookupParticipantId()))
  .flatMap {
    case (Some(lid), Some(pid)) if lid != ledgerId => sys.error()
    case (Some(lid), Some(pid)) if pid != participantId => sys.error()
    case (Some(lid), Some(pid)) => Done
    case (Some(lid), None) => ledgerDao.initializeParticipant()
    case (None, None) => /* initialize with given data */
    case (None, Some(pid)) => ???
  }
```